qcore/tests/test_helpers.py

```diff
@@ -250,3 +251,17 @@ def __repr__(self):
     assert_eq('\'a\'', qcore.safe_repr('a'))
     assert_eq('\'...', qcore.safe_repr('2.192842', max_length=4))
     assert_eq('<n/a: repr...', qcore.safe_repr(TestObject(), max_length=13))
+
+
+def test_marker_object():
+    assert_eq('text', six.text_type(qcore.MarkerObject('text')))
+
+    # bytes should work in py2 but not py3
+    if six.PY2:
+        assert_eq(b'bytes', six.text_type(qcore.MarkerObject(b'bytes')))
```

Comment:
this seems wrong? something is going through text constructor and comes out to be bytes?

In python 2, name is supposed to be unicode and `__str__` just returns that. we should encode into bytes there and also override `__unicode__`.